apps/system/js/browser.js

```diff
@@ -1,9 +1,17 @@
-/* global UrlHelper, AppWindow, BrowserConfigHelper, LazyLoader */
+/* global UrlHelper, AppWindow, BrowserConfigHelper, LazyLoader,
+          SettingsListener */
 
 (function(exports) {
 
   'use strict';
 
+  function handleSettingChange (value) {
+    Browser.privateByDefault = value;
+  }
+
+  SettingsListener.observe('browser.private.default', false,
```

Comment:
I started doing it this way, and realized that I didn't have an instance of a `Browser` in all cases. Specifically in `apps/system/js/wrapper_factory.js`, accessing the static property of the `Browser` class works well.

As it is, `privateByDefault` is a static property and doesn't change from one instance to another.